functions/go/export-terraform/terraformgenerator/resources.go

```diff
@@ -17,43 +17,263 @@ package terraformgenerator
+	value := resource.GetStringFromObject(path...)
+	if len(value) == 0 {
+		return false
+	}
+	boolValue, err := strconv.ParseBool(value)
+	if err != nil {
+		// todo: log the failure
+		return false
+	}
+	return boolValue
+}
+
+// Look up a referenced resource at a given path
+func (resource *terraformResource) GetStringFromObject(path ...string) string {
+	var ref string
+	found, err := resource.Item.Get(&ref, path...)
+	if err != nil || !found {
+		return ""
+	}
+	return ref
+}
+
+// Return if the resource itself should be created
+func (resource *terraformResource) GetOrganization() *terraformResource {
+	if resource.Parent.Kind == "Organization" {
+		return resource.Parent
+	}
+	orgs := resource.resources.getGrouped()["Organization"]
+	if len(orgs) < 1 {
+		sdk.Logf("Failed to fetch organization for %s/%s", resource.Kind, resource.Name)
+		return nil
+	}
+
+	return orgs[0]
+}
+
+// Attach parents and other references to a resource
+func (resource *terraformResource) attachReferences() error {
+	resource.References = make(map[string]*terraformResource)
+	paths := [][]string{
+		{"BillingAccount", "spec", "billingAccountRef", "external"},
+	}
+	for _, path := range paths {
+		kind := path[0]
+		ref := resource.getReferencedResource(kind, path[1:]...)
+		if ref != nil {
+			resource.References[kind] = ref
+		}
+	}
+
+	// attach parents
+	parentKind, parentName, err := resource.getParentRef()
+	if err != nil {
+		sdk.Logf("no parent reference found for %s, %v", resource.Item.Name(), err)
+		return err
+	}
+
+	if parentKind != "" {
+		parentRef, err := resource.resources.getResourceRef(parentKind, parentName, nil)
+		if err != nil {
+			return err
+		}
+		parentRef.Children = append(parentRef.Children, resource)
+		resource.isChild = true
+		resource.Parent = parentRef
+	}
+
+	return nil
+}
+
+// Retrieve a referenced resource from the object spec
+func (resource *terraformResource) getReferencedResource(kind string, path ...string) *terraformResource {
+	name := strings.TrimSpace(resource.GetStringFromObject(path...))
+	if len(name) == 0 {
+		return nil
+	}
+
+	ref, err := resource.resources.getResourceRef(kind, name, nil)
+	if err != nil {
+		sdk.Logf("Failed to retrieve resource reference for %s/%s from %s", kind, name, resource.Name)
+	}
+	ref.Children = append(ref.Children, resource)
+
+	return ref
+}
+
+func (resource *terraformResource) getParentRef(path ...string) (string, string, error) {
+	paths := [][]string{
+		{"Folder", "spec", "folderRef", "name"},
+		{"Folder", "spec", "folderRef", "external"},
+		{"Organization", "spec", "organizationRef", "external"},
+		{"Project", "metadata", "annotations", "cnrm.cloud.google.com/my-project"},
+		{"detect", "spec", "resourceRef", "external"},
+		{"detect", "spec", "resourceRef", "name"},
+	}
+
+	for _, path := range paths {
+		name := resource.GetStringFromObject(path[1:]...)
+		if name == "" {
+			continue
+		}
+
+		kind := path[0]
+		if kind == "detect" {
```

Comment:
Is `detect` a real kind? Or it's something we defined in the function? If the latter, we should define a const for it.